### PR TITLE
Replace embedded OSGi-prefs sources by bundles from Maven-Central

### PR DESCRIPTION
**Pull request by HannesWell**
Similar to the other changes explicitly discussed in https://github.com/eclipse-equinox/equinox.framework/issues/40, this replaces the embedded OSGi-sources of `org.eclipse.equinox.preferences` by the corresponding osgi-bundle from Maven-Central.

As a side effect this updates the versions of
- org.osgi.service.prefs 1.1.1 -> 1.1.2

Since many other plug-ins are likely requiring `o.e.equinox.preferences` and are using the classes in `org.osgi.service.prefs` that bundle is required and re-exported.

### CODE REVIEW

**Comment by HannesWell** [comment]
This requires https://github.com/eclipse-equinox/equinox.bundles/pull/39 to build successfully.
But since we have not yet talked about this particular Plug-in I would like the clarify first if this change should be done or if there are reasons that speak against it.

Additionally some more features have to be updated to keep the sources of `org.osgi.service.prefs` available.

**Review by bjhargrave — COMMENTED** [review]
Since you require the osgi prefs API bundle at `1.1.0`, the import package version should match. It is currently `1.1.1`

**Comment by bjhargrave** [comment]
> This requires #39 to build successfully.

This is PR 39. Perhaps you mean another PR?

**Comment by HannesWell** [comment]
> > This requires #39 to build successfully.
> 
> This is PR 39. Perhaps you mean another PR?

Ups, indeed. I meant: https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/pull/247

> Since you require the osgi prefs API bundle at `1.1.0`, the import package version should match. It is currently `1.1.1`

Thanks for the hint. Shouldn't I remove the Import-Package entirely to not have bot Import-Package and Require-bundle of org.osgi.service.prefs?

File: bundles/org.eclipse.equinox.preferences/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -7,11 +7,11 @@ Bundle-Activator: org.eclipse.core.internal.preferences.Activator
 Bundle-Vendor: %providerName
 Bundle-Localization: plugin
 Require-Bundle: org.eclipse.equinox.common;bundle-version="[3.2.0,4.0.0)",
- org.eclipse.equinox.registry;bundle-version="[3.2.0,4.0.0)";resolution:=optional
+ org.eclipse.equinox.registry;bundle-version="[3.2.0,4.0.0)";resolution:=optional,
+ org.osgi.service.prefs;bundle-version="[1.1.0,1.2.0)";visibility:=reexport
```
Comment (tjwatson) [inline bundles/org.eclipse.equinox.preferences/META-INF/MANIFEST.MF:11]: You need to remove the `Import-Package` for `org.osgi.service.prefs`.  I don't recall what the behavior is supposed to be if you require->reexport a bundle but then import a package that required->reexport bundle also exports.  What should happen if your import wires to a different export from what is required?  Anyway, you should just remove the `Import-Package` regardless.

**Comment by bjhargrave** [comment]
> Shouldn't I remove the Import-Package entirely to not have bot Import-Package and Require-bundle of org.osgi.service.prefs?

I am not sure if Require-Bundle with reexport allows for substitution? @tjwatson? If it does, then having the Import-Package is good since it provides more room for the resolver.

File: bundles/org.eclipse.equinox.preferences/META-INF/MANIFEST.MF (hunk 1 above)
Comment (HannesWell) [reply bundles/org.eclipse.equinox.preferences/META-INF/MANIFEST.MF:11]: Done.
Btw. We had the same topic in PR 26: https://github.com/eclipse-equinox/equinox.bundles/pull/26#discussion_r859800539

**Comment by tjwatson** [comment]
> I am not sure if Require-Bundle with reexport allows for substitution? @tjwatson? If it does, then having the Import-Package is good since it provides more room for the resolver.

I think this is a grey area.  I know the resolver will make sure the class space is consistent for the importing bundle.  But I don't recall what would happen for the bundles requiring the re-exporter (`org.ecipse.equinox.prefs` in this case).  I don't recall if they get wired to the target of the import package wire or directly to the reexported bundle.

**Review by tjwatson — APPROVED** [review]
LGTM

**Comment by HannesWell** [comment]
Thanks everyone!

**Comment by HannesWell** [comment]
For completeness I created/updated the following PRs to include org.osgi.service.prefs into those features that already include `o.e.equinox.preferences`:
- https://github.com/eclipse-equinox/p2/pull/56
- https://github.com/eclipse-platform/eclipse.platform.runtime/pull/45
- https://github.com/eclipse-platform/eclipse.platform.ui/pull/42

**Comment by tjwatson** [comment]
We should move the application admin tests out of `org.eclipse.osgi.tests` and into some dedicated `org.eclipse.equinox.app.tests` project.  Of coarse the test failures in the `org.eclipse.osgi.tests` would have been caught during the this PR build if we had combined `framework` and `bundles` repos also.

**Comment by HannesWell** [comment]
> We should move the application admin tests out of `org.eclipse.osgi.tests` and into some dedicated `org.eclipse.equinox.app.tests` project. Of coarse the test failures in the `org.eclipse.osgi.tests` would have been caught during the this PR build if we had combined `framework` and `bundles` repos also.

Sounds good. Having finer grained test-plugins is probably at all not a bad idea.
Yes combined repos would have helped too, but the merge is not too far away.

**Pull request merged**